docs/internal/release_process.md

```diff
@@ -0,0 +1,45 @@
+Gluecodium release process
+==========================
+
+This document describes the approach to versioning and release process of the Gluecodium process.
+
+Semantic versioning
+-------------------
+
+Gluecodium loosely follows the approach to version numbers known as "semantic versioning". Quoting
+[semver.org](https://semver.org):
+
+> Given a version number MAJOR.MINOR.PATCH, increment the:
+> 1. MAJOR version when you make incompatible API changes,
+> 2. MINOR version when you add functionality in a backwards compatible manner, and
+> 3. PATCH version when you make backwards compatible bug fixes.
+
+Gluecodium versions mostly follow this logic. The exceptions are based on the size of the feature being added:
+* for *very small* features only PATCH version is incremented, even if it's not a bug fix.
+* for *very big* features (e.g. a new generator) the MAJOR version is incremented, even if it's not breaking.
+
+Changelog
+---------
+
+Committed user-visible changes to Gluecodium are (manually) tracked in the [CHANGELOG.md](../../CHANGELOG.md) file. Each
+new change should be briefly described at the top of the file, under `Unreleased` section. Most changes fall either into
+`Bug fixes` subsection or into `Features` subsection. There are also rarer subsections for breaking changes, for
+deprecations, and for removals.
+
+Committed changes that are not user-visible (e.g. refactoring) are not tracked in the changelog.
+
+Release process
+---------------
+
+Gluecodium has no fixed release schedule. Most MINOR and PATCH releases happen on-demand, according to requirements from
+internal (in-company) users of Gluecodium. For bigger features (all MAJOR releases, and some MINOR) the release happens
+as soon as the feature is ready.
+
+Each release raises the current Gluecodium version in a dedicated commit. The same commit should have the current
+`Unreleased` section of the changelog converted into the `Release MAJOR.MINOR.PATCH` (also have a "release date" line
+added there).
+
+After the version change is committed, a release tag needs to be created manually on the
+[Releases](https://github.com/heremaps/gluecodium/releases) page. The convention is to copy-paste the current release
+section from the changelog into the "description" field: this way the users of Gluecodium, who subscribed to release
+notifications, also get the release notes for this release.
```

Comment:
So after release branch is created then Appveyor pick up it automatically, right? It is worth to be mentioned. Also it would be nice to mention somewhere why GitHub actions are not used for releasing

Reply:
> Appveyor 
Did you mean GitHub Actions? If yes, then yes, it will pick it up automatically, but so it will for any other non-release branch, so it does not seems to be worth mentioning. Also, the CI picking up the release branch is not relevant to the release process as there is nothing to build or verify.

> why GitHub actions are not used for releasing
"Because no one ever suggested that before" is not the thing that should be documented. It is the thing to have a ticket created for: #919.